content/en/docs/concepts/workloads/pods/probes.md

```diff
@@ -0,0 +1,166 @@
+
+- **Liveness Probe:** Checks if a container is still running. If the liveness probe fails, Kubernetes restarts the container to recover from issues like deadlocks or crashes.
+- **Readiness Probe:** Determines if a container is ready to handle traffic. If the readiness probe fails, the Pod is removed from the Service's endpoint list, preventing traffic from being routed to it.
+- **Startup Probe:** Ensures that a container has successfully started. It is particularly useful for applications with long initialization times. Startup probes override liveness and readiness probes until they succeed.
+
+---
+
+## Probe Configuration
+
+Probes are configured as part of the Pod specification in YAML manifests. Each probe can define multiple parameters to suit specific application requirements. Below are the details for configuring each type of probe:
+
+#### Key Fields
+- **`path`**: The HTTP endpoint to query.
+- **`port`**: The container port to target.
+- **`initialDelaySeconds`**: Time to wait before starting the probe.
+- **`periodSeconds`**: Interval between probe attempts.
+- **`timeoutSeconds`**: Maximum time to wait for a response.
+- **`failureThreshold`**: Number of consecutive failures before the probe is considered failed.
+- **`successThreshold`**: Number of consecutive successes before the probe is considered successful.
+
+### HTTP Probe
+
+An HTTP probe sends an HTTP GET request to a specific endpoint within the container. This is useful for applications exposing an HTTP server.
+
+#### Example Configuration
+
+```yaml
+livenessProbe:
+  httpGet:
+    path: /healthz
+    port: 8080
+  initialDelaySeconds: 5
+  periodSeconds: 10
+  timeoutSeconds: 2
+  failureThreshold: 3
+  successThreshold: 1
+```
+
+---
+
+### TCP Probe
+
+A TCP probe attempts to establish a TCP connection on a specified port. This is useful for applications that don't use HTTP but expose services on a specific port.
+
+#### Example Configuration
+
+```yaml
+readinessProbe:
+  tcpSocket:
+    port: 3306
+  initialDelaySeconds: 10
+  periodSeconds: 5
+```
+
+#### Key Fields
+- **`tcpSocket`**: Specifies the target port for the TCP check.
+- Other fields like `initialDelaySeconds`, `periodSeconds`, `timeoutSeconds`, `failureThreshold`, and `successThreshold` work similarly to HTTP probes.
+
+---
+
+### Exec Probe
+
+An exec probe runs a command inside the container and determines health based on the exit code.
+
+#### Example Configuration
+
+```yaml
+startupProbe:
+  exec:
+    command:
+      - cat
+      - /tmp/ready
+  initialDelaySeconds: 10
+  periodSeconds: 5
+```
+
+#### Behavior
+- An **exit code of `0`** indicates success.
+- A **non-zero exit code** indicates failure.
+
+---
+
+## gRPC Health Checks
+
+Kubernetes can use the gRPC Health Check Protocol for monitoring gRPC services. The gRPC health check response includes different statuses that can be leveraged for configuring liveliness and readiness probes.
```

Comment:
"gRPC Health Check Protocol" is a term?
Why capitalize the first characters in each words?